docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,408 @@
+[[migrate-to-java-time]]
+=== Java time migration guide
+
+With 7.0, {es} switched from joda time to java time for date-related parsing,
+formatting, and calculations. This guide is designed to help you determine
+if your cluster is impacted and, if so, prepare for the upgrade.
+
+[[java-time-convert-date-formats]]
+=== Convert date formats
+
+To upgrade to 7.0, you'll need to convert any joda-time date formats
+to their java-time equivalents.
+
+To help track this effort, you can prefix java-time date formats with an `8`
+prefix in {es} 6.8 and later versions.
+
+For example, you can change the date format `YYYY-MM-dd` to `8yyyy-MM-dd` to
+indicate the date format uses java time.
+
+{es} treats date formats starting with the `8` prefix differently depending on
+the version:
+
+*6.8*: Date formats with an `8` prefix are handled as java-time formats. Date
+formats without an `8` prefix are treated as joda-time formats. We recommend
+converting these joda-time formats to java-time _before_ upgrading to 7.x.
+
+*7.x and later*: For indices created in 6.x, date formats without an `8` prefix
+are treated as joda-time formats. For indices created in 7.x and later versions,
+all date formats are treated as java-time formats, regardless of whether it
+starts with an `8` prefix.
+
+[[java-time-migration-impacted-features]]
+==== Impacted features
+The switch to java time only impacts custom <<date,`date`>> and
+<<date_nanos,`date_nanos`>> formats.
+
+These formats are commonly used in:
+
+* <<mapping,Index mappings>>
+* <<indices-templates,Index templates>>
+* <<pipeline,Ingest pipelines>>
+
+If you don't use custom date formats, you can skip the rest of this guide.
+Most custom date formats are compatible. However, several require
+an update.
+
+To see if your date format is impacted, use the <<migration-api-deprecation,deprecation info API>>
+or the {kibana-ref}/upgrade-assistant.html[Kibana upgrade assistant]
+
+[[java-time-migration-incompatible-date-formats]]
+==== Incompatible date formats
+Custom date formats containing the following joda-time literals should be
+changed before upgrading.
+
+`Y` (Year of era)::
++
+--
+Replace with `y`.
+
+*Example:*
+`YYYY-MM-dd` should become `yyyy-MM-dd`.
+
+In java time, `Y` is used for
+https://docs.oracle.com/javase/8/docs/api/java/time/temporal/WeekFields.html[week-based year].
+Using `Y` in place of `y` could result in off-by-one errors in year calculation.
+
+
+For pattern `YYYY-ww` and date `2019-01-01T00:00:00.000Z`  will give `2019-01`
+For pattern `YYYY-ww` and date `2018-12-31T00:00:00.000Z`  will give `2019-01` (counter intuitive) because there is >4 days of that week in 2019
```

Comment:
Probably need to rewrap this line as well.
```suggestion
For pattern `YYYY-ww` and date `2018-12-31T00:00:00.000Z`  will give `2019-01` (counter-intuitive) because there is >4 days of that week in 2019
```